app/build.gradle

```diff
@@ -61,6 +61,12 @@ android {
         exclude 'META-INF/maven/com.google.guava/guava/pom.properties'
         exclude 'META-INF/maven/com.google.guava/guava/pom.xml'
     }
+    sourceSets {
+        androidTest {
+            java.srcDirs = ['src/androidTestE2eTest/java',
+                            'src/androidTestServer/java']
```

Comment:
```AppiumUiAutomator2Server``` - it's a ```while(true){}``` test. I think it's a not good idea to include this test in common set.

Reply:
do you know another way to show it in the sources list?

Reply:
What do you mean by "sources list"?
There are two build variants in the project. First is the server, second is e2e tests for the server.

Reply:
I mean this file is not shown by Android Studio if we use the default Gradle settings

Reply:
<img width="916" alt="scr" src="https://user-images.githubusercontent.com/7767781/39615200-3ecfe508-4f75-11e8-8800-b70ab311d29f.png">

Reply:
![image](https://user-images.githubusercontent.com/3530074/39615412-b1805390-4f7e-11e8-8727-aa8312c4cdae.png)

Reply:
it is visible in the project tree, but is not included into Android sources. This is how it looks without that Gradle entry:

![scr2](https://user-images.githubusercontent.com/7767781/39616030-abb2f8aa-4f79-11e8-9110-6914a12ceca4.png)

Reply:
I';ve moved the class to main source set if androidTests is not the best place for it

Reply:
![image](https://user-images.githubusercontent.com/3530074/39616427-05ce2972-4f84-11e8-910c-b3d20660f23c.png)